pkg/downloader/chart_downloader.go

```diff
@@ -117,6 +182,21 @@ func (c *ChartDownloader) DownloadTo(ref, version, dest string) (string, *proven
 	// If provenance is requested, verify it.
 	ver := &provenance.Verification{}
 	if c.Verify > VerifyNever {
+		// If debug is enabled, use the same debug transport for provenance file
+		if debug {
+			dt := &debugTransport{
+				Transport: http.DefaultTransport.(*http.Transport).Clone(),
+				out:       c.Out,
+			}
+			dt.DisableKeepAlives = true
+			c.Options = append(c.Options, getter.WithClient(&http.Client{
```

Comment:
I'm not super familiar with this code, can the user provide a client here? Do you need to possibly work with a client specified by the user?